examples/kdf-example.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#include <mococrw/kdf.h>
+#include <iostream>
+
+using namespace mococrw;
+
+struct KdfData {
+    std::vector<uint8_t> password;
+    size_t outputLength;
+    std::vector<uint8_t> salt;
+    DigestTypes digestType;
+    uint32_t iterations;
+};
+
+void pbkdf2(const struct KdfData &kdfData)
+{
+    auto pbkdf2 = PBKDF2(kdfData.digestType, kdfData.iterations);
+    auto derivedKey = pbkdf2.deriveKey(kdfData.password, kdfData.outputLength, kdfData.salt);
+
+    std::cout << utility::toHex(derivedKey) << std::endl;
+}
+
+void x963kdf(const struct KdfData &kdfData)
+{
+    auto x963kdf = X963KDF(kdfData.digestType);
+    auto derivedKey = x963kdf.deriveKey(kdfData.password, kdfData.outputLength, kdfData.salt);
+
+    std::cout << utility::toHex(derivedKey) << std::endl;
+}
+
+int main(void)
+{
+    struct KdfData kdfData = {
+        utility::fromHex("deadbeef"),
+        512 / 8,
```

Comment:
```suggestion
        Hash::getDigestSize(DigestTypes::SHA512),
```
please avoid manual calcualtion of digest length as we already have a function do that (already fixed the same thing in the ecies implementation)